src/components/NewDatePicker/CalendarPicker/index.js

```diff
@@ -201,11 +201,11 @@ class CalendarPicker extends React.PureComponent {
                         style={styles.flexRow}
                     >
                         {_.map(week, (day, index) => {
-                            const currentDate = moment([currentYearView, currentMonthView, day]);
-                            const isBeforeMinDate = currentDate < moment(this.props.minDate).startOf('day');
-                            const isAfterMaxDate = currentDate > moment(this.props.maxDate).startOf('day');
+                            const currentDate = new Date(currentYearView, currentMonthView, day);
+                            const isBeforeMinDate = currentDate < startOfDay(new Date(this.props.minDate));
+                            const isAfterMaxDate = currentDate > startOfDay(new Date(this.props.maxDate));
                             const isDisabled = !day || isBeforeMinDate || isAfterMaxDate;
-                            const isSelected = moment(this.props.value).isSame(moment([currentYearView, currentMonthView, day]), 'day');
+                            const isSelected = isSameDay(new Date(this.props.value), new Date(currentYearView, currentMonthView, day));
```

Comment:
In `moment` if we pass a `null` day we will get an invalid date but this is not the case with `Date`. This lead to a regression https://github.com/Expensify/App/issues/31793.

```js
moment([2000, 1, null]).format() // 'Invalid date'
new Date(2000, 1, null) // Mon Jan 31 2000 00:00:00 GMT+0100 (Central European Standard Time)
```